authors/apps/articles/views.py

```diff
@@ -0,0 +1,82 @@
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from django.utils.text import slugify
+from rest_framework import status
+from django.utils import timezone
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from .models import Article
+from .serializers import ArticleSerializer
+
+class Articles(APIView):
+    #Route protection
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        articles = Article.objects.all().order_by('createdAt')
+        serializer = ArticleSerializer(articles, many=True)
+        articles_count = len(serializer.data)
+        all_articles = serializer.data
+        articles_list = []
+        for article in all_articles:
```

Comment:
Iterating through articles like this has a negative impact on efficiency as the list grows. We will need to research a better way to do it.  otherwise well done   @AtamaZack